kasa/smart/modules/lightstripeffect.py

```diff
@@ -105,9 +105,11 @@ async def set_effect(
         """
         brightness_module = self._device.modules[Module.Brightness]
         if effect == self.LIGHT_EFFECTS_OFF:
-            effect_dict = dict(self.data["lighting_effect"])
-            effect_dict["enable"] = 0
-        elif effect not in self._effect_mapping:
+            state = self._device.modules[Module.Light].state
+            await self._device.modules[Module.Light].set_state(state)
+            return
```

Comment:
```suggestion
            await self._device.turn_off()
            return
```
Would be simpler.

Reply:
The above suggestion is incorrect, as the PR sets the full state instead of turning the device off.